File: multi/1_Two_sum/Two_Sum.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        int[] sums = new int[2];
        IDictionary<int, int> done = new Dictionary<int, int>();
        for(int i = 0; i < nums.Length; i++){
            if(done.ContainsKey(target-nums[i])){
                sums[0]=done[target-nums[i]];
                sums[1]=i;
                return sums;
            }
            done[nums[i]] = i;
        }
        return sums;
    }
}

File: multi/1_Two_sum/Two_Sum.swift
class Solution {
    func twoSum(_ nums: [Int], _ target: Int) -> [Int] {
    var done = [Int: Int]()
    var sums = [Int]()
    for i in 0...nums.count-1{
        let key = target-nums[i]
        if let val = done[key] {
            sums.append(val)
            sums.append(i)
            return sums
        }
        done[nums[i]] = i
    }
    return [0]
}
}